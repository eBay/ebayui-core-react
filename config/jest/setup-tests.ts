import '@testing-library/jest-dom'

declare var global: typeof globalThis;

if (typeof window.URL.createObjectURL === 'undefined') {
  Object.defineProperty(window.URL, 'createObjectURL', { value: () => {} });
}
