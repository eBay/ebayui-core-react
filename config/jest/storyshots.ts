import { render } from "@testing-library/react";
import initStoryshotsAddon from "@storybook/addon-storyshots";
import type { StoryshotsOptions } from "@storybook/addon-storyshots/dist/ts3.9/api/StoryshotsOptions";

const reactTestingLibrarySerializer = {
    print: (val, serialize, indent) => serialize(val.asFragment()),
    test: (val) => val && val.hasOwnProperty("container"),
};

export function initStoryshots(config: StoryshotsOptions) {
    initStoryshotsAddon({
        ...config,
        renderer: render,
        snapshotSerializers: [reactTestingLibrarySerializer],
    });
}
