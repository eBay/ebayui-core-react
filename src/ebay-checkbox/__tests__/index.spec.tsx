import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { initStoryshots } from '../../../config/jest/storyshots'
import { EbayCheckbox } from '../../index'

describe('<EbayCheckbox>', () => {
    describe('on checkbox-button click', () => {
        it('should fire an event', () => {
            const spy = jest.fn()
            const { getByLabelText, debug } = render(
                <EbayCheckbox aria-label="checkbox" onChange={spy} />
            )
            const input = getByLabelText('checkbox')
            userEvent.click(input);
            expect(spy).toBeCalled()
        })
    })
})

initStoryshots({
    config: ({ configure }) =>
        configure(() => {
            require('./index.stories')
        }, module)
});
