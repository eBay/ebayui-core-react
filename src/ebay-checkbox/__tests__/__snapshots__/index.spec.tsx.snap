// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-checkbox Control value from outside 1`] = `
<DocumentFragment>
  <span
    class="checkbox custom"
    style="align-items: center;"
  >
    <input
      aria-label="custom color checkbox example"
      checked=""
      class="checkbox__control"
      id="checkbox-30"
      type="checkbox"
    />
    <span
      class="checkbox__icon"
      hidden=""
    >
      <svg
        aria-hidden="true"
        class="checkbox__checked icon icon--checkbox-checked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-checkbox-checked"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="checkbox__unchecked icon icon--checkbox-unchecked"
        focusable="false"
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlink:href="#icon-checkbox-unchecked"
        />
      </svg>
    </span>
  </span>
  <label
    class="field__label field__label--end"
    for="checkbox-30"
  >
    Gets disabled after 5 clicks
  </label>
</DocumentFragment>
`;

exports[`Storyshots ebay-checkbox Default checkbox-button 1`] = `
<DocumentFragment>
  <p>
    <span
      class="checkbox"
      style="align-items: center;"
    >
      <input
        class="checkbox__control"
        id="checkbox-11"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-11"
    >
      Default
    </label>
  </p>
  <p>
    <span
      class="checkbox checkbox--large"
      style="align-items: center;"
    >
      <input
        class="checkbox__control"
        id="checkbox-12"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked-large"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked-large"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-12"
    >
      Large
    </label>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-checkbox Disabled checkbox-button 1`] = `
<DocumentFragment>
  <p>
    <span
      class="checkbox"
      style="align-items: center;"
    >
      <input
        class="checkbox__control"
        disabled=""
        id="checkbox-31"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-31"
    >
      Default disabled
    </label>
  </p>
  <p>
    <span
      class="checkbox checkbox--large"
      style="align-items: center;"
    >
      <input
        class="checkbox__control"
        disabled=""
        id="checkbox-32"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked-large"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked-large"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-32"
    >
      Large disabled
    </label>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-checkbox Grouped checkbox-buttons 1`] = `
<DocumentFragment>
  <fieldset>
    <legend>
      Choose an Option
    </legend>
    <span
      class="field"
    >
      <span
        class="checkbox"
        style="align-items: center;"
      >
        <input
          class="checkbox__control"
          id="group-checkbox-1"
          name="checkbox-group"
          type="checkbox"
          value="1"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--checkbox-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--checkbox-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-checkbox-1"
      >
        Option 1
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="checkbox"
        style="align-items: center;"
      >
        <input
          class="checkbox__control"
          id="group-checkbox-2"
          name="checkbox-group"
          type="checkbox"
          value="2"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--checkbox-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--checkbox-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-checkbox-2"
      >
        Option 2
      </label>
    </span>
    <span
      class="field"
    >
      <span
        class="checkbox"
        style="align-items: center;"
      >
        <input
          class="checkbox__control"
          id="group-checkbox-3"
          name="checkbox-group"
          type="checkbox"
          value="3"
        />
        <span
          class="checkbox__icon"
          hidden=""
        >
          <svg
            aria-hidden="true"
            class="checkbox__checked icon icon--checkbox-checked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-checked"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="checkbox__unchecked icon icon--checkbox-unchecked"
            focusable="false"
            height="24px"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <use
              xlink:href="#icon-checkbox-unchecked"
            />
          </svg>
        </span>
      </span>
      <label
        class="field__label field__label--end"
        for="group-checkbox-3"
      >
        Option 3
      </label>
    </span>
  </fieldset>
</DocumentFragment>
`;

exports[`Storyshots ebay-checkbox Selected checkbox-button 1`] = `
<DocumentFragment>
  <p>
    <span
      class="checkbox"
      style="align-items: center;"
    >
      <input
        checked=""
        class="checkbox__control"
        id="checkbox-21"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-21"
    >
      Default
    </label>
  </p>
  <p>
    <span
      class="checkbox checkbox--large"
      style="align-items: center;"
    >
      <input
        checked=""
        class="checkbox__control"
        id="checkbox-22"
        type="checkbox"
        value="123"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked-large"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked-large"
          focusable="false"
          height="64px"
          width="64px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked-large"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-22"
    >
      Large
    </label>
  </p>
</DocumentFragment>
`;

exports[`Storyshots ebay-checkbox Styled checkbox-button 1`] = `
<DocumentFragment>
  <span
    class="checkbox"
  >
    <style>
      
                .custom ~ label { color: green; }
                .custom ~ .checkbox__icon svg { color: green !important; }
            
    </style>
    <span
      class="checkbox custom"
      style="align-items: center;"
    >
      <input
        aria-label="custom color checkbox example"
        class="checkbox__control"
        id="checkbox-30"
        type="checkbox"
      />
      <span
        class="checkbox__icon"
        hidden=""
      >
        <svg
          aria-hidden="true"
          class="checkbox__checked icon icon--checkbox-checked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-checked"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="checkbox__unchecked icon icon--checkbox-unchecked"
          focusable="false"
          height="24px"
          width="24px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlink:href="#icon-checkbox-unchecked"
          />
        </svg>
      </span>
    </span>
    <label
      class="field__label field__label--end"
      for="checkbox-30"
    >
      Custom style
    </label>
  </span>
</DocumentFragment>
`;
